tests/network/vmi_istio.go

```diff
@@ -89,65 +100,103 @@ var _ = SIGDescribe("[Serial] Istio", func() {
 			}()
 
-			By("Create NetworkAttachmentDefinition")
-			nad := generateIstioCNINetworkAttachmentDefinition()
-			_, err = virtClient.NetworkClient().K8sCniCncfIoV1().NetworkAttachmentDefinitions(tests.NamespaceTestDefault).Create(context.TODO(), nad, metav1.CreateOptions{})
-			Expect(err).ShouldNot(HaveOccurred())
-
 			By("Creating VMI")
-			ports := []v1.Port{
-				{Port: svcDeclaredTestPort},
-				{Port: svcUndeclaredTestPort},
-			}
-			vmi = newVMIWithIstioSidecar(ports)
+			vmi = newVMIWithIstioSidecar(vmiPorts)
 			vmi, err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Create(vmi)
 			Expect(err).ShouldNot(HaveOccurred())
 
-			By("Creating k8s service")
-			service := newService()
-			_, err = virtClient.CoreV1().Services(tests.NamespaceTestDefault).Create(context.Background(), service, metav1.CreateOptions{})
-			Expect(err).ShouldNot(HaveOccurred())
-
 			By("Waiting for VMI to be ready")
 			tests.WaitUntilVMIReady(vmi, console.LoginToCirros)
 		})
 		Describe("Inbound traffic", func() {
-			table.DescribeTable("request to VMI", func(port int) {
-				job, err := createJobCheckingVMIReachability(vmi, port)
-				Expect(err).ShouldNot(HaveOccurred())
-
+			checkVMIReachability := func(vmi *v1.VirtualMachineInstance, targetPort int) error {
+				job, err := createJobCheckingVMIReachability(vmi, targetPort)
+				if err != nil {
+					return err
+				}
 				By("Waiting for the job to succeed")
-				Expect(tests.WaitForJobToSucceed(job, 480*time.Second)).To(Succeed())
-			},
-				table.Entry("should reach VMI HTTP server on service declared port", svcDeclaredTestPort),
-				table.Entry("should reach VMI HTTP server on service undeclared port", svcUndeclaredTestPort),
-			)
+				return tests.WaitForJobToSucceed(job, 480*time.Second)
+			}
+
+			Context("With VMI having explicit ports specified", func() {
+				BeforeEach(func() {
+					vmiPorts = explicitPorts
+				})
+				table.DescribeTable("request to VMI should reach HTTP server", func(targetPort int) {
+					Expect(checkVMIReachability(vmi, targetPort)).To(Succeed())
+				},
+					table.Entry("on service declared port on VMI with explicit ports", svcDeclaredTestPort),
+					table.Entry("on service undeclared port on VMI with explicit ports", svcUndeclaredTestPort),
+				)
+			})
+			Context("With VMI having no explicit ports specified", func() {
+				BeforeEach(func() {
+					vmiPorts = []v1.Port{}
+				})
+				table.DescribeTable("request to VMI should reach HTTP server", func(targetPort int) {
+					Expect(checkVMIReachability(vmi, targetPort)).To(Succeed())
+				},
+					table.Entry("on service declared port on VMI with no explicit ports", svcDeclaredTestPort),
+					table.Entry("on service undeclared port on VMI with no explicit ports", svcUndeclaredTestPort),
+				)
+				It("Should not be able to reach service running on Istio restricted port", func() {
```

Comment:
Not directly related to this PR. What if a vm with specified ports has one of the istio restricted port? Maybe we should verify it in the validation webhook when creating the vm/vmi.

Reply:
> What if a vm with specified ports has one of the istio restricted port?

That would disrupt (well, break) the proxy. As you suggest, we could handle this in a webhook.
Initially, I believe it's okay to leave this reponsibility to the user - there is no difference in this matter from using a regular pod with Istio.

Reply:
What would "handle" mean here ? Rejecting the VM creation request ? 

I would vote for something a bit less conservative: log wherever a requested port is listed as a restricted port; possibly fail if all requested ports are in the restricted section.

Reply:
> What would "handle" mean here ? Rejecting the VM creation request ?

Hmm, rejecting was my first idea.
If we only log warning about this, users may not notice that right away and since outbound traffic wouldn't be affected and would be still proxied, this may lead to weird networking issues

Reply:
My point is that it might be a bit overkill to reject a vmi because one port out of the 50 you've requested is reserved.

But I do understand your point. If you know something won't work, why bother creating it ? 

I don't know what's worse.